### DISCUSSION TITLE
Help with Custom Jupyter Kernel Setup in Colab

### DISCUSSION THREAD

**Discussion by sohang3112** [evcxr/evcxr#386] [General]
Hi, I'm working on setting up a custom Jupyter kernel in Colab (trying to replicate the "Open in Colab" button in [evcxr](https://github.com/evcxr/evcxr/blob/main/evcxr_jupyter/README.md)). I've managed to install the kernel, but on the last step (similar to the script in the [evcxr sample notebook](https://github.com/evcxr/evcxr/blob/main/evcxr_jupyter/samples/evcxr_jupyter_tour.ipynb)), the runtime disconnects instead of restarting with the new kernel. 

@wiseaidev @davidlattimore  I understand this issue is unrelated to evcxr, but I was wondering if there are any tutorials or resources on properly setting up a custom kernel within Colab? Any advice would be greatly appreciated!

**Comment by davidlattimore** [comment]
Hi @sohang3112. Unfortunately, I know very little about Colab. I assume from what you've written that you've downloaded the URL referenced in the notebook, unpacked it locally and looked through the script contained within? If you use that script as-is, does it work for then? If it does, then perhaps you can narrow down a particular change you're making that is causing it to stop working.

**Comment by wiseaidev** [comment]
Interesting! I'll try hacking Google Colab this weekend and work on fixing the script!

**Comment by sohang3112** [comment]
@wiseaidev just clarifying that the colab setup in ecvxr is working fine, but I couldn't replicate it for a different custom Jupyter kernel in Colab. The custom kernel is IForth (for Forth language) which I am working on - [Colab notebook](https://colab.research.google.com/github/sohang3112/iforth/blob/master/forth_jupyter_tour.ipynb) using [this kernel setup script](https://github.com/sohang3112/iforth/blob/master/colab_setup_forth_kernel.sh) (last line should restart jupyter but doesn't start again after killing jupyter). If you have the time, please do have a look at these, you might be able to spot what's going wrong in it. 

Thanks in advance 🙂
